hmac.cc

```diff
@@ -42,42 +42,46 @@ struct TestData {
   string key;  // Hex
   string data;  // Hex
   string hash;  // Hex
+  // C_SignInit may complain if the key is too short
+  // (i.e. shorter than HMAC output size).
+  // if short_key is true, failing with CKR_KEY_SIZE_RANGE is allowed.
+  bool short_key;
 };
 
 map<string, vector<TestData> > kTestVectors = {
   // Test vectors from RFC 2202.
   {"MD5-HMAC",
    {{"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b", "4869205468657265",
-     "9294727a3638bb1c13f48ef8158bfc9d"},
+     "9294727a3638bb1c13f48ef8158bfc9d", false},
     {"4a656665", "7768617420646f2079612077616e7420666f72206e6f7468696e673f",
-     "750c783e6ab0b503eaa86e310a5db738"}}},
+     "750c783e6ab0b503eaa86e310a5db738", true}}},
```

Comment:
Hmm, I wonder now whether this should be auto-determined (i.e have `bool short_key = (testcase.key.size() < testdata.data.size());` in the test body) &ndash; wdyt?  (Sorry, should have thought of that before.)

Reply:
That looks good to me, I should have though about that..
Anyway I'll fix that one.